packages/web-api/src/controllers/scan-request-controller.ts

```diff
@@ -48,40 +62,54 @@ export class ScanRequestController extends ApiController {
         });
     }
 
-    private getProcessedRequestData(batchId: string, scanRunRequests: ScanRunRequest[]): ProcessedBatchRequestData {
+    private async getProcessedRequestData(batchId: string, scanRunRequests: ScanRunRequest[]): Promise<ProcessedBatchRequestData> {
         const scanRequestsToBeStoredInDb: ScanRunBatchRequest[] = [];
         const scanResponses: ScanRunResponse[] = [];
 
-        scanRunRequests.forEach(scanRunRequest => {
-            if (Url.tryParseUrlString(scanRunRequest.url) !== undefined) {
-                // preserve GUID origin for a single batch scope
-                const scanId = this.guidGenerator.createGuidFromBaseGuid(batchId);
-                scanRequestsToBeStoredInDb.push({
-                    scanId: scanId,
-                    priority: isNil(scanRunRequest.priority) ? 0 : scanRunRequest.priority,
-                    url: scanRunRequest.url,
-                });
-
-                scanResponses.push({
-                    scanId: scanId,
-                    url: scanRunRequest.url,
-                });
-            } else {
-                scanResponses.push({
-                    url: scanRunRequest.url,
-                    error: WebApiErrorCodes.invalidURL.error,
-                });
-            }
-        });
+        await Promise.all(
+            scanRunRequests.map(async scanRunRequest => {
+                const runRequestValidationResult = await this.validateRunRequest(scanRunRequest);
+                if (runRequestValidationResult.valid) {
+                    // preserve GUID origin for a single batch scope
+                    const scanId = this.guidGenerator.createGuidFromBaseGuid(batchId);
+                    scanRequestsToBeStoredInDb.push({
+                        scanId: scanId,
+                        priority: isNil(scanRunRequest.priority) ? 0 : scanRunRequest.priority,
+                        url: scanRunRequest.url,
+                    });
+
+                    scanResponses.push({
+                        scanId: scanId,
+                        url: scanRunRequest.url,
+                    });
+                } else {
+                    scanResponses.push({
+                        url: scanRunRequest.url,
+                        error: runRequestValidationResult.error,
+                    });
+                }
+            }),
+        );
 
         return {
             scanRequestsToBeStoredInDb: scanRequestsToBeStoredInDb,
             scanResponses: scanResponses,
         };
     }
-}
 
-interface ProcessedBatchRequestData {
-    scanRequestsToBeStoredInDb: ScanRunBatchRequest[];
-    scanResponses: ScanRunResponse[];
+    private async validateRunRequest(scanRunRequest: ScanRunRequest): Promise<RunRequestValidationResult> {
```

Comment:
looks like this function doesnt have to be async & return promise.

Reply:
updated